apps/04TowersOfHanoi.js

```diff
@@ -13,43 +13,80 @@ var stacks = {
   c: []
 };
 
+
 function printStacks() {
   console.log("a: " + stacks.a);
   console.log("b: " + stacks.b);
   console.log("c: " + stacks.c);
 }
 
-function movePiece() {
-  // Your code here
-
+function movePiece(startStack, endStack) {
+  // Pop and push a piece!
+  var piece = stacks[startStack].pop();
+  stacks[endStack].push(piece);
 }
 
-function isLegal() {
-  // Your code here
-
+function isLegal(startStack, endStack) {
+  // Check if endStack has something in it.
+  if (stacks[endStack].length < 1) {
+    return true;
+  }
+  // Check if the last number in endStack is larger than the piece you are moving from startStack. 
+  // Or if this is a legal move.
+  else if ((stacks[startStack][(stacks[startStack].length-1)]) < (stacks[endStack][(stacks[endStack].length-1)])) {
+    return true;
+  }
+  // If not, this is an illegal move.
+  else {
+    console.log("Illegal move! Try again!");
+    return false;
+  }
 }
 
 function checkForWin() {
-  // Your code here
-
+  // Check if the b stack or c stack has a .length of 4, if yes, user wins!
+  if (stacks.c.length > 3){
+    console.log("You Won!!");
+    return true;
+  }
+  else if (stacks.b.length > 3){
+    console.log("You Won!!");
+    return true;
+  }
+  else{
+    console.log("Keep Going!");
+    return false;
+  }
 }
 
 function towersOfHanoi(startStack, endStack) {
-  // Your code here
-
+  // If isLegal returns true, move a piece!
+  if (isLegal(startStack, endStack)){
+    movePiece(startStack,endStack);
```

Comment:
Nicely job evaluating `isLegal()` before allowing your code to run `movePiece()`.